tests/src/test/java/org/eclipse/hono/tests/Tenants.java

```diff
@@ -66,14 +80,42 @@ public static Tenant createTenantForTrustAnchor(final X509Certificate cert, fina
      * @param state The enabled state.
      */
     public static void setAdapterEnabled(final Tenant tenant, final String type, final Boolean state) {
+        setAdapterEnabled(tenant, type, state, false);
+    }
+
+    /**
+     * Create a new adapter section and set the enabled state.
+     * 
+     * @param tenant The tenant to process.
+     * @param type   The adapter type to configure.
+     * @param state  The enabled state.
+     * @param authRequired  The device-authentication-required property.
+     */
+    public static void setAdapterEnabled(final Tenant tenant, final String type, final Boolean state,
+            final boolean authRequired) {
+        setAdapterEnabled(tenant, type, state, authRequired, null);
+    }
+
+    /**
+     * Creates a new adapter section and set the given properties for the adapter.
+     * 
+     * @param tenant        The tenant to process.
+     * @param type          The adapter type to configure.
+     * @param state         The enabled state.
+     * @param authRequired  The device-authentication-required property.
+     * @param extensions    The extensions properties of the tenant.
+     */
+    public static void setAdapterEnabled(final Tenant tenant, final String type, final Boolean state, final boolean authRequired, final Map<String, Object> extensions) {
```

Comment:
this method's name is misleading. In fact, it *adds* a new configuration for an adapter based on the given properties.
FMPOV something like *addAdapterConfiguration* would be more appropriate. But then, why isn't this a method of the `Tenant` object itself?

Reply:
> this method's name is misleading. In fact, it adds a new configuration for an adapter based on the given properties.

Yes, i agree!

> But then, why isn't this a method of the Tenant object itself?

IMHO, we should keep the `Tenant` class as simple as possible and avoid populating it with utility methods (as we currently do in the `TenantObject` class). If anybody would read the device registry API, especially the `Tenant` component schema, one would see a direct relationship between the schema and the Java class.

IMHO, we should not add utility methods to that class, to the point that it becomes hard to see how it even relates to the schema defined in the API specification. I would suggest creating a specialized utility class, that defines only static methods, for adding such helper methods. WDYT?

Reply:
I would (partially) agree if we would be generating the value objects from the OpenAPI spec. But since we don't, I feel that this more like a case of an anemic value object. There is no problem at all to see the relation between these objects and the OpenAPI spec's schema definition. They have the same name and properties. Adding methods that make using them much easier poses no problem at all FMPOV.
The confusion with the `TenantObject` only stems from the fact that we have been using the same object for both, the Management API and the Tenant API.

@dejanb @ctron WDYT?

Reply:
@sophokles73 @Alfusainey I generally agree that we can add these methods that deal with adapters directly to the `Tenant` class. But maybe for this use case, we can just define a new constructor for the `Adapter` class and `Tenant.addAdapter()` method, so we could do things like

`tenant.addAdapter(new Adapter(...))`

Reply:
thanks, i updated the PR to allow creating an adapter instance for a given type